### ISSUE TITLE
in refit() : ValueError: Floating-point under-/overflow occurred at epoch #1

### ISSUE THREAD

**Issue by Motorrat** [automl/auto-sklearn#163]
The refit() method crashed.

``````
    c.refit(X.values, y)
  File "/x/Redly/anaconda3/lib/python3.5/site-packages/autosklearn/automl.py", line 421, in refit
    model.fit(X.copy(), y.copy())
  File "/x/Redly/anaconda3/lib/python3.5/site-packages/autosklearn/pipeline/base.py", line 63, in fit
    self.fit_estimator(X, y, **fit_params)
  File "/x/Redly/anaconda3/lib/python3.5/site-packages/autosklearn/pipeline/base.py", line 138, in fit_estimator
    self.pipeline_.steps[-1][-1].fit(X, y, **fit_params)
  File "/x/Redly/anaconda3/lib/python3.5/site-packages/autosklearn/pipeline/components/classification/passive_aggressive.py", line 25, in fit
    self.iterative_fit(X, y, n_iter=1)
  File "/x/Redly/anaconda3/lib/python3.5/site-packages/autosklearn/pipeline/components/classification/passive_aggressive.py", line 57, in iterative_fit
    self.estimator.partial_fit(X, y, classes=np.unique(y))
  File "/x/Redly/anaconda3/lib/python3.5/site-packages/sklearn/linear_model/passive_aggressive.py", line 141, in partial_fit
    coef_init=None, intercept_init=None)
  File "/x/Redly/anaconda3/lib/python3.5/site-packages/sklearn/linear_model/stochastic_gradient.py", line 377, in _partial_fit
    sample_weight=sample_weight, n_iter=n_iter)
  File "/x/Redly/anaconda3/lib/python3.5/site-packages/sklearn/linear_model/stochastic_gradient.py", line 426, in _fit_binary
    sample_weight)
  File "/x/Redly/anaconda3/lib/python3.5/site-packages/sklearn/linear_model/stochastic_gradient.py", line 278, in fit_binary
    est.power_t, est.t_, intercept_decay)
  File "sklearn/linear_model/sgd_fast.pyx", line 404, in sklearn.linear_model.sgd_fast.plain_sgd (sklearn/linear_model/sgd_fast.c:4919)
  File "sklearn/linear_model/sgd_fast.pyx", line 697, in sklearn.linear_model.sgd_fast._plain_sgd (sklearn/linear_model/sgd_fast.c:7569)
ValueError: Floating-point under-/overflow occurred at epoch #1. Scaling input data with StandardScaler or MinMaxScaler might help
```.
``````

**Comment by mfeurer** [comment]
This is unfortunate, and we actually don't have any strategy to deal with that. Here are some possible ideas on how this could be prevented:

1. Put a try/catch around the refit, and shuffle your data if it crashes. If you are lucky, the crash only appears because of bad ordering effects of the data.
2. Print the ensemble before actually calling refit. If it only has a weight of 0.02/0.04/0.06 there should be no harm in removing that model and increasing the weight of a different model.

Is this the first time this error happens to you? If not, we should put an automated procedure to deal with such crashes into auto-sklearn.

**Comment by Motorrat** [comment]
This has happened a few times already. My goal is to have a fully automated "fire and forget" pipeline with auto-sklearn so I would appreciate if this could be caught internally. Data may have many millions of rows so if this _linear_ part of the flow crashes we'd have to restart it. The fit() is running in parallel so a crash in one process there does not stop the whole pipeline.

**Comment by Motorrat** [comment]
Just for the record - I have wrapped the refit() method with try/except and just restarted it. It has failed the same the second time.

**Comment by mfeurer** [comment]
Did you shuffle your data or change the seed when you called refit() the 2nd time?

**Comment by Motorrat** [comment]
I did not shuffle when it was failing two times. Now I have added the shuffle but don't have statistics yet. I will update here if the crash happens again.

**Comment by Motorrat** [comment]
So far the shuffle seems to have solved the problem, but I am not sure I have tested with all datasets where it might have failed before.

**Comment by mfeurer** [comment]
Thank you for this confirmation. We will add a check in the refit-method and then change the seed or shuffle the data if fitting the algorithm failed.

**Comment by mfeurer** [comment]
Was fixed in 3b2eb47641e71a1362704cd419dab50087f38dac. Will be available in the next release.
